Subject: [v2,3/3] ARM: OMAP2+: am43xx: Add context offset for dss hwmod
Date: Sat, 15 Feb 2014 02:07:35 -0000
From: Dave Gerlach <d-gerlach@ti.com>

Add context_offs to the DSS hwmod for am43xx as the driver uses the
context loss count to determine whether or not to restore context to the
module. Also add prcm flag indicating this bit is valid. Without this the
display does not return after a suspend cycle.

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>

---
arch/arm/mach-omap2/omap_hwmod_33xx_data.c |    2 ++
 arch/arm/mach-omap2/prcm43xx.h             |    3 +++
 2 files changed, 5 insertions(+)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index 978356d..b298d6d 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -1921,7 +1921,9 @@ static struct omap_hwmod am43xx_dss_core_hwmod = {
 	.prcm = {
 		.omap4 = {
 			.clkctrl_offs = AM43XX_CM_PER_DSS_CLKCTRL_OFFSET,
+			.context_offs = AM43XX_PRM_RM_PER_DSS_CONTEXT,
 			.modulemode   = MODULEMODE_SWCTRL,
+			.flags = HWMOD_AM437X_HAS_CONTEXT_LOSS_BIT,
 		},
 	},
 };
diff --git a/arch/arm/mach-omap2/prcm43xx.h b/arch/arm/mach-omap2/prcm43xx.h
index f5344ab..55c41bb 100644
--- a/arch/arm/mach-omap2/prcm43xx.h
+++ b/arch/arm/mach-omap2/prcm43xx.h
@@ -25,6 +25,9 @@
 #define AM43XX_PRM_WKUP_INST				0x2000
 #define AM43XX_PRM_DEVICE_INST				0x4000
 
+/* PRM.PRM_PER offsets */
+#define AM43XX_PRM_RM_PER_DSS_CONTEXT			0x0A24
+
 /* RM RSTCTRL offsets */
 #define AM43XX_RM_PER_RSTCTRL_OFFSET			0x0010
 #define AM43XX_RM_GFX_RSTCTRL_OFFSET			0x0010
